Scrapping_part_3/scrapping_part3/scrapping_part3/middlewares.py

```diff
@@ -0,0 +1,46 @@
+from stem import Signal
+from stem.control import Controller
+from scrapy.downloadermiddlewares.httpproxy import HttpProxyMiddleware
+from stem.util.log import get_logger
+
+from .settings import REQUESTS_PER_SAME_TOR_IDENTITY
```

Comment:
You should access settings through Scrapy framework, not directly import it
See https://docs.scrapy.org/en/latest/topics/settings.html#how-to-access-settings